cmd/entrypoints/clusterresource.go

```diff
@@ -41,12 +48,37 @@ func getClusterResourceController(ctx context.Context, scope promutils.Scope, co
 		panic(err)
 	}
 
-	clientSet, err := admin.ClientSetBuilder().WithConfig(admin.GetConfig(ctx)).Build(ctx)
-	if err != nil {
-		panic(err)
+	var adminDataProvider interfaces.FlyteAdminDataProvider
+	if configuration.ClusterResourceConfiguration().IsStandaloneDeployment() {
+		clientSet, err := admin.ClientSetBuilder().WithConfig(admin.GetConfig(ctx)).Build(ctx)
+		if err != nil {
+			panic(err)
+		}
+		adminDataProvider = impl.NewAdminServiceDataProvider(clientSet.AdminClient())
+	} else {
+		dbConfigValues := configuration.ApplicationConfiguration().GetDbConfig()
+		dbLogLevel := gormLogger.Silent
+		if dbConfigValues.Debug {
+			dbLogLevel = gormLogger.Info
+		}
+		dbConfig := repositoryConfig.DbConfig{
+			BaseConfig: repositoryConfig.BaseConfig{
+				LogLevel: dbLogLevel,
+			},
+			Host:         dbConfigValues.Host,
+			Port:         dbConfigValues.Port,
+			DbName:       dbConfigValues.DbName,
+			User:         dbConfigValues.User,
+			Password:     dbConfigValues.Password,
+			ExtraOptions: dbConfigValues.ExtraOptions,
+		}
```

Comment:
I have seen this been constructed in different places. Is it possible to restrict this object construction at one place as the DBConfig is the common one being used by all consumers including admin and some init containers

Reply:
oh good call, in fact this already exists as a method 😅 updated